server/src/main/java/org/elasticsearch/script/ScriptStats.java

```diff
@@ -68,6 +72,28 @@ public ScriptStats(
         );
     }
 
+    public static ScriptStats merge(ScriptStats first, ScriptStats second) {
+        var mergedContextStats = List.<ScriptContextStats>of();
+
+        if (first.contextStats.isEmpty() == false || second.contextStats.isEmpty() == false) {
+            var mapToCollectMerges = new HashMap<String, ScriptContextStats>();
```

Comment:
Nit: maybe rename it to `mapToCollectMergedStats` or something in that direction. No strong feelings, just an idea.